modules/View.js

```diff
@@ -698,22 +721,79 @@
      * @method invokeDetached
      */
     invokeDetached: function() {
-      // No need to check if child views are actually detached, because if parent is detached, children are detached.
       if (this.__attachedCallbackInvoked) {
         this._detached();
         this.__attachedCallbackInvoked = false;
       }
-      _.each(this.__childViews, function(view) {
-        view.invokeDetached();
-      });
-      _.each(this.__sharedViews, function(view) {
-        view.invokeDetached();
+      _.each(this.getTrackedViews(), function(view) {
+        // No need to check if tracked view is actually detached from the DOM as long as they are attached to the parent and the if parent is detached,
+        // then the tracked view is detached.
+        if (view.isAttachedToParent()) {
+          view.invokeDetached();
+        }
       });
     },
 
     /************** Private methods **************/
 
     /**
+     * Gets the hash from id to views of the correct views given the options.
+     * @param [options={}] {Object}  Optional options.
+     *   @param [options.shared=false] {Boolean} The view is a shared view instead of a child view
+     *                                           (shared views are not disposed when the parent is disposed)
+     * @method __getTrackedViewsHash
+     * @return the object map cotaining either the shared or child views keyed by their cid
+     * @private
+     */
+    __getTrackedViewsHash: function(options) {
```

Comment:
should this work the same as getTrackedViews() with an option for shared an child separately (and no options returns both)?